### ISSUE TITLE
syntaxnet: parser_eval --graph_builder=greedy ,Assign requires shapes of both tensors to match. lhs shape= [32,69] rhs shape= [400,69]

### ISSUE THREAD

**Issue by 0YuanZhang0** [tensorflow/tensorflow#11737]
Error: 
![image](https://user-images.githubusercontent.com/21216639/28559200-238f40a6-7148-11e7-90c6-0f426bfe8426.png)

I use GPU to train;
when greedy training, I can train model and parser eval with --graph_builder=structured successful, but parser_eval with --graph_builder=greedy failed;
here is my configuration: 
"bazel-bin/syntaxnet/parser_trainer" \
  --arg_prefix=brain_parser \
  --batch_size=400 \
  --compute_lexicon \
  --decay_steps=4400 \
  --graph_builder=greedy \
  --hidden_layer_sizes=1024,1024 \
  --learning_rate=0.08 \
  --momentum=0.9 \
  --seed=4 \
  --output_path=$TMP_DIR \
  --task_context=$TMP_DIR/context \
  --training_corpus=training-corpus \
  --tuning_corpus=tuning-corpus \
  --params=$PARAMS \
  --num_epochs=25 \
  --report_every=1000 \
  --checkpoint_every=10000 \
  --logtostderr

"bazel-bin/syntaxnet/parser_eval" \
  --task_context=$TMP_DIR/brain_parser/greedy/$PARAMS/context \
  --hidden_layer_sizes=1024,1024 \
  --input=dev-corpus \
  --output=stdout \
  --arg_prefix=brain_parser \
  --graph_builder=greedy \
  --model_path=$TMP_DIR/brain_parser/greedy/$PARAMS/model \
  --logtostderr \
  > $TMP_DIR/greedy-out

"bazel-bin/syntaxnet/parser_eval" \
  --task_context=$TMP_DIR/context \
  --hidden_layer_sizes=1024,1024 \
  --beam_size=1 \
  --input=dev-corpus \
  --output=stdout \
  --arg_prefix=brain_parser \
  --graph_builder=structured \
  --model_path=$TMP_DIR/brain_parser/greedy/$PARAMS/model \
  --logtostderr \
  > $TMP_DIR/struct-beam1-out

**Comment by reedwm** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
